MakeItGif

```diff
@@ -1,26 +1,36 @@
 #!/bin/bash
 if [ $# -lt 3 ]
-	then
-		echo "MakeItGif v1.0.0 , Makes gif out of videos using ffmpeg."
-		echo "Copyright (c) 2017 antony jr."
-		echo " "
-		echo "Usage: MakeItGif [VIDEO FILE] [SCALE] [OUTPUT]"
-		exit -1
+    then
+        echo "MakeItGif v1.0.0 , Makes gif out of videos using ffmpeg."
+        echo "Copyright (c) 2017 antony jr."
+        echo " "
+        echo "Usage: MakeItGif [VIDEO FILE] [SCALE] [OUTPUT]"
+        exit -1
 fi
 
+#Check If ffmpeg Is Installed
+if ! [ -x "$(command -v ffmpeg)" ]; then
+printf "ffmpeg not installed , Installing ffmpeg\n"
+
+apt-get update > /dev/null 2>&1
+apt-get install ffmpeg -y > /dev/null 2>&1
+
+else
+
+printf "ffmpeg processing...\n"
 echo "INPUT VIDEO :: "$1
-echo "SCALED 	  :: "$2
+echo "SCALED       :: "$2
 echo "OUTPUT GIF  :: "$3
 
-ffmpeg -y -i $1 -vf fps=10,scale=$2:-1:flags=lanczos,palettegen ~/palette.png
+ffmpeg -y -t 3 -i $1 -vf fps=10,scale=$2:-1:flags=lanczos,palettegen ~/palette.png
```

Comment:
why only the first 3 seconds of the video ?

Reply:
For me it not work without -t 3

Reply:
but this would make the script only make the the gif out of the first 3 secs of the video? we can just remove it and make the users edit it if they face problems like you , because this is just a small script and I'm not planning to make it user-friendly. your thoughts?